packages/beacon-node/src/chain/rewards/syncCommitteeRewards.ts

```diff
@@ -0,0 +1,58 @@
+import {CachedBeaconStateAllForks, CachedBeaconStateAltair} from "@lodestar/state-transition";
+import {ValidatorIndex, allForks, altair} from "@lodestar/types";
+import {ForkName, SYNC_COMMITTEE_SIZE} from "@lodestar/params";
+import {routes} from "@lodestar/api";
+
+export type SyncCommitteeRewards = routes.beacon.SyncCommitteeRewards;
+
+export async function computeSyncCommitteeRewards(
+  block: allForks.BeaconBlock,
+  preState: CachedBeaconStateAllForks,
+  filters?: (ValidatorIndex | string)[]
+): Promise<SyncCommitteeRewards> {
+  const fork = preState.config.getForkName(block.slot);
+  if (fork === ForkName.phase0) {
+    throw Error("Cannot get sync rewards as phase0 block does not have sync committee!");
+  }
+
+  const altairBlock = block as altair.BeaconBlock;
+  const preStateAltair = preState as CachedBeaconStateAltair;
+  const {index2pubkey} = preStateAltair.epochCtx;
+
+  // Bound committeeIndices in case it goes beyond SYNC_COMMITTEE_SIZE just to be safe
+  const committeeIndices = preStateAltair.epochCtx.currentSyncCommitteeIndexed.validatorIndices.slice(
+    0,
+    SYNC_COMMITTEE_SIZE
+  );
+  const {syncParticipantReward} = preStateAltair.epochCtx;
+  const {syncCommitteeBits} = altairBlock.body.syncAggregate;
+
+  // Use balance of each committee as starting point such that we cap the penalty to avoid balance dropping below 0
+  const balances: Map<ValidatorIndex, {val: number}> = new Map(
+    committeeIndices.map((i) => [i, {val: preStateAltair.balances.get(i)}])
+  ); // Use val for convenient way to increment/decrement balance
```

Comment:
This comment seems misplaced, maybe could add new type `BalanceRecord` (on top of file, not exported) and add comment there?

I think we should also prefer `as type` on line 37 instead of non-null-assertion